internal/jobs/twitterx/scraper.go

```diff
@@ -324,3 +492,122 @@ func (s *TwitterXScraper) lookupUserByID(userID string) (string, error) {
 		return "", fmt.Errorf("API user lookup failed with status: %d", resp.StatusCode)
 	}
 }
+
+// GetProfileByID fetches complete user profile information by user ID
+func (s *TwitterXScraper) GetProfileByID(userID string) (*TwitterXProfileResponse, error) {
+	logrus.Infof("Looking up profile for user with ID: %s", userID)
+
+	// Construct endpoint URL with user fields
+	endpoint := fmt.Sprintf("users/%s?user.fields=id,name,username,description,location,url,verified,protected,created_at,profile_image_url,profile_banner_url,public_metrics", userID)
+
+	// Make the request
+	resp, err := s.twitterXClient.Get(endpoint)
+	if err != nil {
+		logrus.Errorf("Error looking up profile: %v", err)
+		return nil, fmt.Errorf("error looking up profile: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Read response body
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		logrus.Errorf("Error reading response body: %v", err)
+		return nil, fmt.Errorf("error reading response body: %w", err)
+	}
+
+	// Check response status first
+	switch resp.StatusCode {
+	case http.StatusOK:
+		// Parse into structured type
+		var profileResp TwitterXProfileResponse
+		if err := json.Unmarshal(body, &profileResp); err != nil {
+			logrus.Errorf("Error parsing response: %v", err)
+			return nil, fmt.Errorf("error parsing response: %w", err)
+		}
+
+		// Check for API errors
+		if len(profileResp.Errors) > 0 {
+			logrus.Errorf("API error: %s (code: %d)", profileResp.Errors[0].Message, profileResp.Errors[0].Code)
+			return nil, fmt.Errorf("API error: %s", profileResp.Errors[0].Message)
+		}
+
+		logrus.Infof("Successfully retrieved profile for user %s (@%s)", profileResp.Data.Name, profileResp.Data.Username)
+		return &profileResp, nil
+	case http.StatusUnauthorized:
+		return nil, fmt.Errorf("invalid API key")
```

Comment:
For all of these constant errors (that have no interpolations), use global vars created with `errors.New` (e.g.
```
var (
   ErrInvalidApiKey = errors.New("invalid API key")
   ErrRateLimitExceeded = errors.New("rate limit exceeded")
   ErrUserNotFound = errors.New("user not found")
)
```